middleware/queue/source/forward/handle.cpp

```diff
@@ -926,25 +931,93 @@ namespace casual
                   }
                } // state
 
+
                namespace dead
                {
-                  namespace detail::transform
+                  namespace detail::push::faked
                   {
-                     auto dequeue( forward::state::pending::Dequeue& pending)
+                     void reply( const forward::state::pending::Enqueue& pending)
                      {
-                        ipc::message::group::dequeue::forget::Request message{ common::process::handle()};
+                        Trace trace{ "queue::forward::service::local::handle::dead::detail::push::faked::reply"};
+                        log::line( verbose::log, "pending: ", pending);
+
+                        ipc::message::group::enqueue::Reply message;
                         message.correlation = pending.correlation;
-                        return message;
+
+                        log::line( verbose::log, "message: ", message);
+                        
+                        ipc::device().push( message);
                      }
+                     
+                  } // detail::push::faked
 
-                     auto enqueue( forward::state::pending::Enqueue& pending)
+                  namespace detail
+                  {
+                     void dead_event( State& state, common::process::compare_equal_to_handle auto handle)
                      {
-                        ipc::message::group::enqueue::Reply message;
-                        message.correlation = pending.correlation;
-                        return message;
+                        // Note: We rely on others to send us "error replies" for the following 
+                        //  * pending.service.calls : service-manager, gateway::outbound
+                        //  * pending.transaction.(commit|rollback) : transaction-manager
+
+                        log::line( verbose::log, "state.pending: ", state.pending);
+
+                        static constexpr auto is_forward = []( auto id) { return [ id]( auto& forward){ return forward == id;};};
+
```

Comment:
I built this branch with g++-13 (with my changes for gcc-13 feature/1.7/main rebased to this branch). 
I got a warning/error:
`compile: /export/home/guo/c_1.7/casual-1.7/middleware/queue/source/forward/handle.cpp
error:
/export/home/guo/c_1.7/casual-1.7/middleware/queue/source/forward/handle.cpp: In instantiation of ‘void casual::queue::forward::local::{anonymous}::handle::dead::detail::dead_event(casual::queue::forward::State&, auto:153) [with auto:153 = casual::common::strong::Type<int, casual::common::strong::detail::integral::policy<int, casual::common::strong::process::tag, -1> >]’:
/export/home/guo/c_1.7/casual-1.7/middleware/queue/source/forward/handle.cpp:1029:43:   required from here
/export/home/guo/c_1.7/casual-1.7/middleware/queue/source/forward/handle.cpp:964:47: error: variable ‘is_forward’ set but not used [-Werror=unused-but-set-variable]
  964 |                         static constexpr auto is_forward = []( auto id) { return [ id]( auto& forward){ return forward == id;};};
      |                                               ^~~~~~~~~~
/export/home/guo/c_1.7/casual-1.7/middleware/queue/source/forward/handle.cpp: In instantiation of ‘void casual::queue::forward::local::{anonymous}::handle::dead::detail::dead_event(casual::queue::forward::State&, auto:153) [with auto:153 = casual::common::strong::Type<casual::common::Uuid, casual::common::strong::ipc::tag>]’:
/export/home/guo/c_1.7/casual-1.7/middleware/queue/source/forward/handle.cpp:1041:43:   required from here
/export/home/guo/c_1.7/casual-1.7/middleware/queue/source/forward/handle.cpp:964:47: error: variable ‘is_forward’ set but not used [-Werror=unused-but-set-variable]
cc1plus: all warnings being treated as errors

aborting due to errors`
I can't say if this is a legitimate warning or not. I looked arounf a little in bugzilla for gcc/g++ and there seemed to be som issues having to do with  constexpr and parameters and lambda captures in the gcc-9 to gcc-11 timeframe that has been corrected.
I removed static and added "explicit" capture of "is_forward" to the invalidate_queue_forward and invalidate_service_forward lambdas. Then code builds without warings. And unit tests run.

Reply:
You should be able to create a constexpr-static non capturing lambda and it behaves just as a free function. Maybe it needs to be `inline` ?

Reply:
I (!) think it is a compiler bug. Turning off the warning is a possibility, but it may be useful and correct in other cases. As far as I know this is the only place the warning is produced with casual.

For some reason the compiler misses that the variable "static constexpr is_forward" is used by  the lambdas invalidate_queue_forward and invalidate_service_forward. Some of the problems they fixed in earlier g++ versions seemed to have to do with the compiler missing to "mark" the use of some variables on some paths through the compiler. There might be one more for the case when a static constexpr lambda is used by another lambda... I will try to reproduce in a small code example. Perhaps not easy. I do not know  how much context is needed...

Reply:
Try to use it without the constexpr. That should also work, but then the compiler needs to guard the initialization for concurrent threads.

Reply:
Already tried. Same compiler warning. Turning off warning or remove static and capture (by value is what I tried) is_forward variable seems to work.

Reply:
High probability for a compiler bug. I'll say we lay low with g++13 for a while. At least until we chose to use it in our build pipeline.